src/main/java/org/apache/maven/plugins/shade/resource/ReproducibleResourceTransformer.java

```diff
@@ -0,0 +1,37 @@
+package org.apache.maven.plugins.shade.resource;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.plugins.shade.relocation.Relocator;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * Copy of original interface necessary to MSHADE-363_old-plugin IT.
```

Comment:
This comment gives me the feeling that this interface exists for the mentioned IT only. Does it?

Reply:
I don't understand what to do with this comment: probably it could be improved.
Copying this interface from maven-shade-plugin 3.2.4 to the transformer extension source code is required because the extension uses it but will be run in the context of older maven-shade-plugin releases that don't provide it.
don't hesitate to propose better code comment...

Reply:
oh, I switched `src/main/java/.../ReproducibleResourceTransformer.java` content with `src/test/java/.../ReproducibleResourceTransformer.java`
will replace files...